autogen/agentchat/groupchat.py

```diff
@@ -36,6 +36,20 @@ class GroupChat:
         When set to True and when a message is a function call suggestion,
         the next speaker will be chosen from an agent which contains the corresponding function name
         in its `function_map`.
+    - custom_speaker_selection_func: pass a function to customize the speaker selection. Default is None.
+        custom_speaker_selection_func:
+            Parameters:
+                - last_speaker: Agent
+                    The last speaker in the group chat.
+                - agents: List[Agents]
+                    All agents in the group chat.
+                - messages: List[Dict]
+                    All past messages in the group chat.
+            Returns:
+                - Agent
+                    The selected speaker.
+        The return agent should be one of the agents in the group chat.
+        If the return agent is not valid, we will use default to the method specified in `speaker_selection_method`.
```

Comment:
What about using `speaker_selection_method` instead of creating a new kwarg?
If you want to keep the behavior of falling back to a str-based selection method, you can allow the return type to be a str.
The parameters defined here is contradictory to line 406.